dell/xps/15-9570/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, ... }:
+
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+  ];
+
+  # TODO: boot loader
+  boot.loader.systemd-boot.enable = lib.mkDefault true;
+  boot.loader.efi.canTouchEfiVariables = lib.mkDefault true;
+
+  nix.buildCores = lib.mkDefault 12;
```

Comment:
Maybe we should make this the default in NixOS:

```
nix.buildCores = lib.mkDefault 0;
```

This sets cores equal to the number of hardware threads in a system.